packages/website/blog/2024-03-01-parser-options-project-service.md

```diff
@@ -0,0 +1,298 @@
+
+- "Full cold start" of linting the entire project
+- "Partial cold start" of seeing editor suggestions after opening the project in VS Code
+- "Warm start" of seeing editor suggestions after changing a file in VS Code
+
+See [typescript-eslint/project-apis-performance-comparison](https://github.com/typescript-eslint/project-apis-performance-comparison) for details on the performance measurements.
+
+### Full Cold Start
+
+These measurements were taken by running `npm run lint` on the respective projects.
+
+| API                | 👶 Tiny | 🧒 Small | 🧑 Medium | 🧓 Large |
+| ------------------ | ------- | -------- | --------- | -------- |
+| Legacy Program     |         |          |           |          |
+| 🆕 Project Service |         |          |           |          |
+| 𝚫 Delta            |         |          |           |          |
+
+### Partial Cold Start
+
+These measurements were taken by recording restarting VS Code on the respective projects with a single file open.
+
+| API                | 👶 Tiny | 🧒 Small | 🧑 Medium | 🧓 Large |
+| ------------------ | ------- | -------- | --------- | -------- |
+| Legacy Program     |         |          |           |          |
+| 🆕 Project Service |         |          |           |          |
+| 𝚫 Delta            |         |          |           |          |
+
+### Warm Start
+
+These measurements were taken by recording VS Code on the respective projects with a single file open and modifying that file.
+
+| API                | 👶 Tiny | 🧒 Small | 🧑 Medium | 🧓 Large |
+| ------------------ | ------- | -------- | --------- | -------- |
+| Legacy Program     |         |          |           |          |
+| 🆕 Project Service |         |          |           |          |
+| 𝚫 Delta            |         |          |           |          |
+
+## Next Steps
+
+### Giving Feedback
+
+We'd love to hear from you on how this option works for you.
+Does it live up to what we've promised, and/or does it have bugs we haven't fixed yet?
+Please do post in the [Community Feedback: Project Service APIs](https://github.com/typescript-eslint/typescript-eslint/discussions/8030) GitHub Discussions post on how it goes for you.
+
+For support in setting up the new APIs, feel free to ask on [the typescript-eslint Discord](https://discord.gg/FSxKq8Tdyg)'s `#project-service` channel.
+We'd be happy to help you try out `parserOptions.projectService`.
+
+### Long Term Vision
+
+Our hope is that the Project Service API becomes the standard way to work with typed linting over the next few major versions.
+Our intent is to roll it out according to the following rough timeline:
+
+- **v7**: Rename `parserOptions.EXPERIMENTAL_useProjectService` to `parserOptions.projectService`
+- **v8**: Rename `parserOptions.project` to something like `parserOptions.DEPRECATED_legacyProjectProgram` and rename `parserOptions.projectService` to `parserOptions.project`
+- **v9**: Remove `parserOptions.DEPRECATED_legacyProjectProgram`
```

Comment:
The one complicated thing about this new setup is that it makes anything like #4359 really hard.
Instead of things "just work"ing because we can parallelise automatically based on config - instead we need to do config discovery on the main thread before we can parallelise.
Just makes things a bit more complex to deal with. Not that it's a blocker for this feature given that a custom CLi is still a pipe dream...  just something to note for future.

Reply:
Hmm yeah good point. I suppose there's no need to explicitly state a removal in this blog post. I'll just remove this bit.